hstream/src/HStream/Server/Handler.hs

```diff
@@ -887,10 +891,17 @@ streamingFetchHandler ServerContext{..} (ServerBiDiRequest _ streamRecv streamSe
         (
           \info@SubscribeRuntimeInfo{..} -> do
             let unackedRecordIds = filterUnackedRecordIds recordIds sriAckedRanges sriWindowLowerBound
-            Log.info $ Log.buildInt (V.length unackedRecordIds) <> " records need to be resend"
-            if V.null unackedRecordIds
-               then return info
-               else doResend info unackedRecordIds
+            if V.null unackedRecordIds || V.null sriStreamSends
```

Comment:
unackedRecords should be resend later even if no valid streamSends now.

Reply:
fixed